server/controllers/conversation.js

```diff
@@ -0,0 +1,167 @@
+
+
+const User = require( "../models/User" );
+const asyncHandler = require( "express-async-handler" );
+const Conversation = require( "../models/conversation" );
+const Message = require( "../models/message" );
+const Profile = require( "../models/Profile" );
+
+// @route Post /create
+// @desc create conversation
+// @access Public
+exports.createConversation = asyncHandler( async ( req, res, next ) => {
+  const userId = await User.findById( req.user.id );
+  if ( !userId )
+  {
+    res.status( 401 );
+    throw new Error( "Not authorized" );
+  }
+  const senderId = req.user.id;
+  const { recipientId, text, conversationId } = req.body;
```

Comment:
Additionally, we can also validate the request body parameters and send a `400` if they're missing 

Ditto where applicable (same goes for path params)

Reply:
added an if clause checks it the req.body parameter is empty or not